datastore/common/common.go

```diff
@@ -1,11 +1,118 @@
 package common
 
 import (
+	"fmt"
+	"log"
 	"os"
+	"strconv"
+	"strings"
+	"time"
 
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+var (
+	dynamicTime bool // whether the valid time range is considered dynamic or static
+	loTimeSecs, hiTimeSecs int64
+)
+
+// initValidTimeRange initializes dynamicTime, loTimeSecs, and hiTimeSecs from environment
+// variables DYNAMICTIME, LOTIME, and HITIME.
+//
+// The valid time range is defined like this if dynamicTime is true:
+//
+//     [now - loTimeSecs, now - hiTimeSecs]
+//
+// where {lo|hi}TimeSecs is converted directly as integer seconds from {LO|HI}TIME.
+//
+// If dynamicTime is false, the valid time range is defined like this:
+//
+//     [loTimeSecs, hiTimeSecs]
+//
+// where {lo|hi}TimeSecs is either converted directly as integer seconds from {LO|HI}TIME, or
+// converted indirectly from {LO|HI}TIME represented as an ISO-8601 datetime of the exact form
+// 2023-10-10T00:00:00Z.
+//
+// NOTE: if any errors are encountered, warnings will be printed and fallback values will be used.
+func initValidTimeRange() {
+	dynTime0 := strings.ToLower(Getenv("DYNAMICTIME", "true"))
+	dynamicTime = ((dynTime0 != "false") && (dynTime0 != "no") && (dynTime0 != "0"))
+
+	// define function that derives a valid int64 time range component from env.var
+	// 'name', with a default value of 'defaultVal', and with the option of
+	// expressing the value as an ISO-8601 datetime as a fallback.
+	getSecs := func(name string, defaultVal int64, allowIso8601 bool) int64 {
+		val0 := strings.ToLower(Getenv(name, fmt.Sprintf("%d", defaultVal)))
+		var val int64
+		var err error
+
+		// first attempt to extract directly from int64
+		if val, err = strconv.ParseInt(val0, 10, 64); err != nil {
```

Comment:
Isn't this clearer in this case?
```suggestion

		// first attempt to extract directly from int64
		val, err := strconv.ParseInt(val0, 10, 64)
		if err != nil {
```

Reply:
Agreed, will fix!